builder/amazon/ebsvolume/step_snapshot_ebs_volumes_test.go

```diff
@@ -0,0 +1,180 @@
+package ebsvolume
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"sync"
+	"testing"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/request"
+
+	//"github.com/aws/aws-sdk-go/service/ec2"
+	"github.com/aws/aws-sdk-go/service/ec2"
+	"github.com/aws/aws-sdk-go/service/ec2/ec2iface"
+	"github.com/hashicorp/packer-plugin-sdk/multistep"
+	"github.com/hashicorp/packer-plugin-sdk/packer"
+	"github.com/hashicorp/packer/builder/amazon/common"
+)
+
+// Define a mock struct to be used in unit tests for common aws steps.
+type mockEC2Conn struct {
+	ec2iface.EC2API
+	Config *aws.Config
+
+	// Counters to figure out what code path was taken
+	copyImageCount       int
+	describeImagesCount  int
+	deregisterImageCount int
+	deleteSnapshotCount  int
+	waitCount            int
+
+	lock sync.Mutex
+}
+
+func (m *mockEC2Conn) CreateSnapshot(input *ec2.CreateSnapshotInput) (*ec2.Snapshot, error) {
```

Comment:
You'd created a mock connection already, but you hadn't implemented the functions on the interface that get called by your step; this is why you were getting crashes.